### ISSUE TITLE
Global installation fails

### ISSUE THREAD

**Issue by WaldemarEnns** [churchtools/changelogger#25]
# Problem
When i try to install the package globally, i get conflicts with packages of my laravel installation:

**Log**
```console
composer global require churchtools/changelogger
Changed current directory to /Users/enux/.composer
Using version ^0.4.0 for churchtools/changelogger
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for churchtools/changelogger ^0.4.0 -> satisfiable by churchtools/changelogger[v0.4.0].
    - Conclusion: remove illuminate/container v6.0.1
    - Conclusion: don't install illuminate/container v6.0.1
    - churchtools/changelogger v0.4.0 requires laravel-zero/framework 5.8.* -> satisfiable by laravel-zero/framework[v5.8.0, v5.8.1, v5.8.2, v5.8.3, v5.8.4, v5.8.5].
    - laravel-zero/framework v5.8.0 requires illuminate/container 5.8.* -> satisfiable by illuminate/container[v5.8.0, v5.8.11, v5.8.12, v5.8.14, v5.8.15, v5.8.17, v5.8.18, v5.8.19, v5.8.2, v5.8.20, v5.8.22, v5.8.24, v5.8.27, v5.8.28, v5.8.29, v5.8.3, v5.8.30, v5.8.31, v5.8.32, v5.8.33, v5.8.34, v5.8.35, v5.8.36, v5.8.4, v5.8.8, v5.8.9].
    - laravel-zero/framework v5.8.1 requires illuminate/container 5.8.* -> satisfiable by illuminate/container[v5.8.0, v5.8.11, v5.8.12, v5.8.14, v5.8.15, v5.8.17, v5.8.18, v5.8.19, v5.8.2, v5.8.20, v5.8.22, v5.8.24, v5.8.27, v5.8.28, v5.8.29, v5.8.3, v5.8.30, v5.8.31, v5.8.32, v5.8.33, v5.8.34, v5.8.35, v5.8.36, v5.8.4, v5.8.8, v5.8.9].
    - laravel-zero/framework v5.8.2 requires illuminate/container 5.8.* -> satisfiable by illuminate/container[v5.8.0, v5.8.11, v5.8.12, v5.8.14, v5.8.15, v5.8.17, v5.8.18, v5.8.19, v5.8.2, v5.8.20, v5.8.22, v5.8.24, v5.8.27, v5.8.28, v5.8.29, v5.8.3, v5.8.30, v5.8.31, v5.8.32, v5.8.33, v5.8.34, v5.8.35, v5.8.36, v5.8.4, v5.8.8, v5.8.9].
    - laravel-zero/framework v5.8.3 requires illuminate/container 5.8.* -> satisfiable by illuminate/container[v5.8.0, v5.8.11, v5.8.12, v5.8.14, v5.8.15, v5.8.17, v5.8.18, v5.8.19, v5.8.2, v5.8.20, v5.8.22, v5.8.24, v5.8.27, v5.8.28, v5.8.29, v5.8.3, v5.8.30, v5.8.31, v5.8.32, v5.8.33, v5.8.34, v5.8.35, v5.8.36, v5.8.4, v5.8.8, v5.8.9].
    - laravel-zero/framework v5.8.4 requires illuminate/container 5.8.* -> satisfiable by illuminate/container[v5.8.0, v5.8.11, v5.8.12, v5.8.14, v5.8.15, v5.8.17, v5.8.18, v5.8.19, v5.8.2, v5.8.20, v5.8.22, v5.8.24, v5.8.27, v5.8.28, v5.8.29, v5.8.3, v5.8.30, v5.8.31, v5.8.32, v5.8.33, v5.8.34, v5.8.35, v5.8.36, v5.8.4, v5.8.8, v5.8.9].
    - laravel-zero/framework v5.8.5 requires illuminate/container 5.8.* -> satisfiable by illuminate/container[v5.8.0, v5.8.11, v5.8.12, v5.8.14, v5.8.15, v5.8.17, v5.8.18, v5.8.19, v5.8.2, v5.8.20, v5.8.22, v5.8.24, v5.8.27, v5.8.28, v5.8.29, v5.8.3, v5.8.30, v5.8.31, v5.8.32, v5.8.33, v5.8.34, v5.8.35, v5.8.36, v5.8.4, v5.8.8, v5.8.9].
    - Can only install one of: illuminate/container[v5.8.0, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.11, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.12, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.14, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.15, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.17, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.18, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.19, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.2, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.20, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.22, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.24, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.27, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.28, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.29, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.3, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.30, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.31, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.32, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.33, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.34, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.35, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.36, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.4, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.8, v6.0.1].
    - Can only install one of: illuminate/container[v5.8.9, v6.0.1].
    - Installation request for illuminate/container (locked at v6.0.1) -> satisfiable by illuminate/container[v6.0.1].


Installation failed, reverting ./composer.json to its original content.
```

Is there already a fix for this? Or any method i can avoid this error? Thank you :)

**Comment by obstschale** [comment]
I will take a look at this later. 

Alternatively, you can use the PHAR file directly. Download the lastest Version here: https://github.com/churchtools/changelogger/releases/download/v0.4.0/changelogger-0.4.0.phar
And execute it using `php`. 

Third Option would be using [Phive](https://github.com/phar-io/phive). This is an tool, which downloads Phars. The benefits are:

* No dependency conflicts, because every tool uses its own PHAR
* Security feature: It verifies the checksum for you.

```sh
phive install --target bin/ --copy https://github.com/churchtools/changelogger/releases/download/v0.4.0/changelogger-0.4.0.phar
```

**Comment by WaldemarEnns** [comment]
> I will take a look at this later.
> 
> Alternatively, you can use the PHAR file directly. Download the lastest Version here: https://github.com/churchtools/changelogger/releases/download/v0.4.0/changelogger-0.4.0.phar
> And execute it using `php`.
> 
> Third Option would be using [Phive](https://github.com/phar-io/phive). This is an tool, which downloads Phars. The benefits are:
> 
> * No dependency conflicts, because every tool uses its own PHAR
> * Security feature: It verifies the checksum for you.
> 
> ```shell
> phive install --target bin/ --copy https://github.com/churchtools/changelogger/releases/download/v0.4.0/changelogger-0.4.0.phar
> ```

Thank you for the quick answer. Since it works, I am using the phar file for now 👍

**Comment by obstschale** [comment]
@WaldemarEnns could you provide your global `composer.json` file here? So I can test with your dependeny tree.

**Comment by obstschale** [comment]
Should be fixed with v0.5.0
